File: languages/go/entity/identityV2/integer_attribute.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type IntegerAttribute struct {
	Attribute
}

func NewIntegerAttribute(key string, value int64) IntegerAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return IntegerAttribute{
		Attribute: a,
	}
}

func NewIntegerAttributeFromProto(s *proto.IntegerAttributeV2) IntegerAttribute {
	if s == nil {
		return IntegerAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return IntegerAttribute{
		Attribute: a,
	}
}

func (s IntegerAttribute) ToProto() *proto.IntegerAttributeV2 {
	return &proto.IntegerAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(int64),
	}
}

File: languages/python/bloock/entity/identity_v2/integer_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class IntegerAttribute(Attribute):
    @staticmethod
    def from_proto(a: proto.IntegerAttributeV2) -> IntegerAttribute:
        return IntegerAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.IntegerAttributeV2:
        return proto.IntegerAttributeV2(
            id=self.id,
            value=self.value,
        )
